src/vma/dev/ring_eth_cb.cpp

```diff
@@ -47,8 +47,8 @@
 #define MAX_MP_WQES		(20) // limit max used memory
 #define MIN_MP_WQES		(4)
 
-ring_eth_cb::ring_eth_cb(int if_index,
-			 vma_cyclic_buffer_ring_attr *cb_ring, ring *parent):
+ring_eth_cb::ring_eth_cb(int if_index, vma_cyclic_buffer_ring_attr *cb_ring,
+			 vma_memory_descriptor *mem_desc, ring *parent):
```

Comment:
this was not done due to arch, @alexvm 
vma_cyclic_buffer_ring_attr is a template that describe a general ring, adding address makes it specific and not a template.

Reply:
I understand but it is possible to introduce internal struct ring_desc that describes ring_eth_cb. This struct can include template info and iovec data. See example at https://github.com/Mellanox/libvma/commit/83596eaaac7b6ced44414cbb86256c06db42c8e0

Constructor can look as 
```
ring_eth_cb(int if_index, vma_cyclic_buffer_ring_attr *cb_ring,
			 struct ring_eth_cb_desc {
vma_cyclic_buffer_ring_attr *cb_ring;
vma_memory_descriptor *mem_desc;
}, ring *parent)
```
This approach allows to pass additional data in case it is needed in the future w/o changing declaration.